paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowController.kt

```diff
@@ -326,44 +327,56 @@ internal class DefaultFlowController @Inject internal constructor(
             is PaymentSelection.GooglePay -> launchGooglePay(state)
             is PaymentSelection.Link,
             is PaymentSelection.New.LinkInline -> confirmLink(paymentSelection, state)
-            is PaymentSelection.New.GenericPaymentMethod -> {
-                if (paymentSelection.paymentMethodCreateParams.typeCode == PaymentMethod.Type.BacsDebit.code) {
-                    BacsMandateData.fromPaymentSelection(paymentSelection)?.let { data ->
-                        bacsMandateConfirmationLauncher.launch(
-                            data = data,
-                            appearance = getPaymentAppearance()
-                        )
-                    } ?: run {
-                        paymentResultCallback.onPaymentSheetResult(
-                            PaymentSheetResult.Failed(
-                                BacsMandateException(
-                                    type = BacsMandateException.Type.MissingInformation
-                                )
-                            )
-                        )
-                    }
-                } else {
-                    confirmPaymentSelection(paymentSelection, state)
-                }
-            }
-            is PaymentSelection.New,
-            null -> confirmPaymentSelection(paymentSelection, state)
-            is PaymentSelection.Saved -> {
-                if (paymentSelection.paymentMethod.type == PaymentMethod.Type.SepaDebit &&
-                    viewModel.paymentSelection?.hasAcknowledgedSepaMandate == false
-                ) {
-                    // We're legally required to show the customer the SEPA mandate before every payment/setup.
-                    // In the edge case where the customer never opened the sheet, and thus never saw the mandate,
-                    // we present the mandate directly.
-                    sepaMandateActivityLauncher.launch(
-                        SepaMandateContract.Args(
-                            merchantName = state.config.merchantDisplayName
+            is PaymentSelection.ExternalPaymentMethod -> ExternalPaymentMethodHandler.confirm(
+                paymentResultCallback::onPaymentSheetResult
+            )
+            is PaymentSelection.New.GenericPaymentMethod -> confirmGenericPaymentMethod(paymentSelection, state)
+            is PaymentSelection.New, null -> confirmPaymentSelection(paymentSelection, state)
+            is PaymentSelection.Saved -> confirmSavedPaymentMethod(paymentSelection, state)
+        }
+    }
+
+    private fun confirmSavedPaymentMethod(
```

Comment:
extracted some functions out of confirm, because I hit a lint error about it being too long